src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drv.c

```diff
@@ -59,6 +59,49 @@ static void xocl_mb_connect(struct xocl_dev *xdev);
 static void xocl_mailbox_srv(void *arg, void *data, size_t len,
 	u64 msgid, int err, bool sw_ch);
 
+static void xocl_mb_read_p2p_addr(struct xocl_dev *xdev)
+{
+	struct pci_dev *pdev = xdev->core.pdev;
+	struct mailbox_req *mb_req = NULL;
+	struct mailbox_p2p_bar_addr *mb_p2p = NULL;
+	size_t mb_p2p_len, reqlen;
+	int ret = 0;
+	size_t resplen = sizeof(ret);
+
+	mb_p2p_len = sizeof(struct mailbox_p2p_bar_addr);
+	reqlen = sizeof(struct mailbox_req) + mb_p2p_len;
+	mb_req = vzalloc(reqlen);
+	if (!mb_req) {
+		userpf_err(xdev, "dropped request (%d), mem alloc issue\n",
+				MAILBOX_REQ_READ_P2P_BAR_ADDR);
+		return;
+	}
+
+	mb_req->req = MAILBOX_REQ_READ_P2P_BAR_ADDR;
+	mb_p2p = (struct mailbox_p2p_bar_addr *)mb_req->data;
+	mb_p2p->p2p_bar_len = pci_resource_len(pdev, xdev->p2p_bar_idx);
+	mb_p2p->p2p_bar_addr = pci_resource_start(pdev, xdev->p2p_bar_idx);
+
+	mutex_lock(&xdev->dev_lock);
+	if (!list_is_singular(&xdev->ctx_list)) {
+		/* We should have one context for ourselves. */
+		BUG_ON(list_empty(&xdev->ctx_list));
+		userpf_err(xdev, "device is in use, can't update p2p in mgmtpf\n");
+		ret = -EBUSY;
+	}
+	mutex_unlock(&xdev->dev_lock);
+	if (ret < 0)
+		return;
+
+	ret = xocl_peer_request(xdev, mb_req, sizeof(struct mailbox_req),
```

Comment:
This call should be  xocl_peer_request(xdev, mb_req, req_len, ... ). Please update it.